storage/innobase/xtrabackup/src/xbcloud/xbcloud.cc

```diff
@@ -1038,10 +1038,16 @@ bool xbcloud_put(Object_store *store, const std::string &container,
 @return true in case of success or false otherwise */
 bool chunk_name_to_file_name(const std::string &chunk_name,
                              std::string &file_name, my_off_t &idx) {
-  if (chunk_name.size() < 22 && chunk_name[chunk_name.size() - 21] != '.') {
+  if (chunk_name.size() < 22 || chunk_name[chunk_name.size() - 21] != '.') {
     /* chunk name is invalid */
     return false;
   }
+  auto slash_pos = chunk_name.find('/');
+  if (slash_pos && chunk_name.size() - slash_pos < 22) {
+    /* '/' must be placed left side from '.' */
+    return false;
+  }
+
```

Comment:
I more interested to know about your done file.

1. Why do you do it?
2. Do you upload 'done' file for every chunk or file?
3. Did you modify xbcloud logic to do it? If you are doing externally, do you do it after the backup is uploaded to S3?